String truncation warnings with newer GCC

**Describe the bug**
Ubuntu has released 20.04 LTS which includes GCC 9.3.0.  When using this compiler it implements a much stricter (and often over-zealous) checking of string ops.

For instance: 
```
In file included from /usr/include/string.h:495,
                 from /home/joe/code/cfecfs/github/tools/cFS-GroundSystem/Subsystems/cmdUtil/cmdUtil.c:33:
In function ‘strncpy’,
    inlined from ‘main’ at /home/joe/code/cfecfs/github/tools/cFS-GroundSystem/Subsystems/cmdUtil/cmdUtil.c:413:13:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10: error: ‘__builtin_strncpy’ specified bound 10 equals destination size [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘strncpy’,
    inlined from ‘main’ at /home/joe/code/cfecfs/github/tools/cFS-GroundSystem/Subsystems/cmdUtil/cmdUtil.c:407:13:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10: error: ‘__builtin_strncpy’ specified bound 32 equals destination size [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**To Reproduce**
Build code with default config using GCC 9.3.0, with optimization enabled and full warnings.

**Expected behavior**
Code should build cleanly.

**System observed on:**
Ubuntu 20.04 LTS 64 bit.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/Subsystems/cmdUtil/cmdUtil.c
+++ b/Subsystems/cmdUtil/cmdUtil.c
@@ -404,13 +404,15 @@ int main(int argc, char *argv[]) {
         {
         case 'H':
             printf("Host: %s\n",(char *)optarg);
-            strncpy(CommandData.HostName, optarg, HOSTNAME_SIZE);
+            strncpy(CommandData.HostName, optarg, HOSTNAME_SIZE-1);
+            CommandData.HostName[HOSTNAME_SIZE-1] = 0;
             CommandData.GotHostName = 1;
             break;
 
         case 'P':
             printf("Port: %s\n",(char *)optarg);
-            strncpy(CommandData.PortNum, optarg, PORTNUM_SIZE);
+            strncpy(CommandData.PortNum, optarg, PORTNUM_SIZE-1);
+            CommandData.PortNum[PORTNUM_SIZE-1] = 0;
             CommandData.GotPortNum = 1;
             break;
 

